src/System.Runtime.Extensions/src/System/Environment.Win32.cs

```diff
@@ -208,14 +208,185 @@ private unsafe static char[] GetEnvironmentCharArray()
 
         private static string GetFolderPathCore(SpecialFolder folder, SpecialFolderOption option)
         {
+            // We're using SHGetKnownFolderPath instead of SHGetFolderPath as SHGetFolderPath is
+            // capped at MAX_PATH.
+            //
+            // Because we validate both of the input enums we shouldn't have to care about CSIDL and flag
+            // definitions we haven't mapped. If we remove or loosen the checks we'd have to account
+            // for mapping here (this includes tweaking as SHGetFolderPath would do).
+            //
+            // The only SpecialFolderOption defines we have are equivalent to KnownFolderFlags.
+
+            string folderGuid = null;
```

Comment:
Nit: can you remove the ```= null```?  Then the compiler should warn if any path in the switch doesn't assign it.